### PR TITLE
B enhancement/gh-669-allowance import create year if not exist

### PR DESCRIPTION
**Pull request by danilo996**

### CODE REVIEW

**Comment by danilo996** [comment]
#669

File: src/main/java/eu/execom/hawaii/batch/AllowanceProcessor.java [hunk 1]
```diff
@@ -98,11 +93,31 @@ private int assignValue(double importedHours) {
...
+  private int calculateNegativeValue(double importedHours) {
+    return calculateHours(Math.abs(importedHours)) * -1;
+  }
+
+  private double roundToLowerValue(double importedHours, double modulus) {
+    return importedHours - modulus;
+  }
+
+  private double roundToHigherValue(double importedHours, double modulus) {
+    return (importedHours + HALF_DAY) - modulus;
```
Comment (mariobasic) [inline src/main/java/eu/execom/hawaii/batch/AllowanceProcessor.java]: Please remove brackets.

File: src/main/java/eu/execom/hawaii/batch/AllowanceProcessor.java [hunk 2]
```diff
@@ -98,11 +93,31 @@ private int assignValue(double importedHours) {
...
-    var result = modulus < 2.5 ? importedHours - modulus : (importedHours + HALF_DAY) - modulus;
+    double modulus = importedHours % HALF_DAY;
+    double result = modulus < MODULUS_BOUNDARY ?
+        roundToLowerValue(importedHours, modulus) :
+        roundToHigherValue(importedHours, modulus);
 
     return (int) Math.round(result);
   }
 
+  private int calculateNegativeValue(double importedHours) {
```
Comment (mariobasic) [inline src/main/java/eu/execom/hawaii/batch/AllowanceProcessor.java]: This method should be above `calculateHours(double importedHours)` method above.

File: src/main/java/eu/execom/hawaii/batch/AllowanceProcessor.java [hunk 3]
```diff
@@ -98,11 +93,31 @@ private int assignValue(double importedHours) {
    * client, so there is no point in having excess amount of hours at users disposal, when user
    * can't use it. Users allowance will be or slightly increased or slightly decreased.
    */
+  private int assignValue(double importedHours) {
+    double modulus = importedHours % HALF_DAY;
+    int calculatedHours = importedHours < 0 ? calculateNegativeValue(importedHours) : calculateHours(importedHours);
+
+    return modulus == 0 ? (int) importedHours : calculatedHours;
```
Comment (mariobasic) [inline src/main/java/eu/execom/hawaii/batch/AllowanceProcessor.java]: If `modulus` is `0` we are returning `importedHours`, that means that calculation of `calculatedHours` at line 98 was unnecessary. Suggestion: 
```
    if (importedHours % HALF_DAY == 0) {
       return (int) importedHours;
    } else {
      return importedHours < 0 ? calculateNegativeValue(importedHours) : calculateHours(importedHours);
    }
```

**Pull request merged**